### DISCUSSION TITLE
The file '/Admin/ControlPanel/ControlBar.ascx' does not exist.

### DISCUSSION THREAD

**Discussion by trouble2** [dnnsoftware/Dnn.Platform#4854] [Q&A]
## Description of bug
I installed a clean DNN 9.9.0 installation on Azure with:
- dutch language
- empty portal template

After installtion I clicked on go to the site and got a 500 error.
In the log file it said something like:
The file '/Admin/ControlPanel/ControlBar.ascx' does not exist.
So I checked the hostsettings table for the controlpanel entry and indeed it showed 
'/Admin/ControlPanel/ControlBar.ascx' 

After I changed it to 
'DesktopModules/admin/Dnn.PersonaBar/UserControls/PersonaBarContainer.ascx'

I was back in business...

## Steps to reproduce
List the precise steps to reproduce the bug:

1. Install a fresh DNN 9.6.1 (I did this on Azure, but I think it does not matter)
2. Select dutch as language
3. Select the Empty website template
4. Install and check what happens

## Current behavior
Installation went ok, but the site gave a 500 error

## Expected behavior
Well....you know, a working site might be nice...

## Error information
2021-03-02 14:27:35.520+00:00 [dw0sdwk00009R][D:17][T:21][FATAL] DotNetNuke.Framework.PageBase - An error has occurred while loading page.
System.Web.HttpException (0x80004005): The file '/Admin/ControlPanel/ControlBar.ascx' does not exist.
   at System.Web.UI.Util.CheckVirtualFileExists(VirtualPath virtualPath)
   at System.Web.Compilation.BuildManager.GetVPathBuildResultInternal(VirtualPath virtualPath, Boolean noBuild, Boolean allowCrossApp, Boolean allowBuildInPrecompile, Boolean throwIfNotFound, Boolean ensureIsUpToDate)
   at System.Web.Compilation.BuildManager.GetVPathBuildResultWithNoAssert(HttpContext context, VirtualPath virtualPath, Boolean noBuild, Boolean allowCrossApp, Boolean allowBuildInPrecompile, Boolean throwIfNotFound, Boolean ensureIsUpToDate)
   at System.Web.UI.TemplateControl.LoadControl(VirtualPath virtualPath)
   at DotNetNuke.UI.ControlUtilities.LoadControl[T](TemplateControl containerControl, String ControlSrc)
   at DotNetNuke.UI.Skins.Skin.InjectControlPanel()
   at DotNetNuke.UI.Skins.Skin.OnInit(EventArgs e)
   at System.Web.UI.Control.InitRecursive(Control namingContainer)
   at System.Web.UI.Control.AddedControl(Control control, Int32 index)
   at System.Web.UI.ControlCollection.Add(Control child)
   at DotNetNuke.Framework.DefaultPage.OnInit(EventArgs e)
   at System.Web.UI.Control.<InitRecursiveAsync>d__225.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Util.WithinCancellableCallbackTaskAwaitable.WithinCancellableCallbackTaskAwaiter.GetResult()
   at System.Web.UI.Page.<ProcessRequestMainAsync>d__523.MoveNext()

## Affected version
* [x] 09.09.00 latest supported release

**Comment by sleupold** [comment]
might be an issue of the "Blank Website" template, or do you experience the same with "Default Website" template?

**Comment by mitchelsellers** [comment]
A few questions on this, as I've seen this before in Azure a few times.

1. What type of resources were you using?  And what was their Size?  (For example P1V3 AppService & S2 SQL)
2. Did you experience a timeout from Azure, or DNN during the installation?
3. Do you remember the total duration of installation time?

Azure PaaS offerings have a hard-set timeout of 230 seconds of inactivity, which can translate to longer given the DNN status updates, but often if installing on low-grade stuff (S1 AppService or S0 DB) it is possible to have issues.

**Comment by trouble2** [comment]
Install was a bit slow indeed, though it went trough in the end without any errors...
I'll try again on a local installation...

**Comment by trouble2** [comment]
Well, looks like that was the issue, on my local installation it went trough fine.
Though you might want to remove that entry in the database/ install scripts anyway to prevent this from happening just in case..
I believe the Admin/ControlPanel/ControlBar.ascx was for DNN 8, so on a clean install for DNN 9 it would be better to remove it all together

**Comment by mitchelsellers** [comment]
For DNN 10.x we are looking to "roll forward' the schema script, until we do that it isn't exactly possible.

I'm curious, what scaling configuration were you using in Azure?  (We have done 30+ installs in Azure in the last month without any issues, so just trying to find the scale point that is an issue so we can possibly document)

**Comment by trouble2** [comment]
I think I did the install using: 
Gen5, 2 vCores, 32 GB storage, zone redundant disabled

**Comment by enfJoao** [comment]
+1 for this issue and the solution above.
Installed on a shared webhost, with no errors.

**Comment by stale** [comment]
We have detected this issue has not had any activity during the last 90  days. That could mean this issue is no longer relevant and/or nobody has  found the necessary time to address the issue. We are trying to keep the  list of open issues limited to those issues that are relevant to the  majority and to close the ones that have become 'stale' (inactive). If no  further activity is detected within the next 14 days, the issue will be  closed automatically.
If new comments are are posted and/or a solution (pull request) is  submitted for review that references this issue, the issue will not be  closed. Closed issues can be reopened at any time in the future. Please remember those participating in this open source project are volunteers  trying to help others and creating a better DNN Platform for all. Thank  you for your continued involvement and contributions!

**Comment by Tychodewaard** [comment]
A few discussions regarding this topic on the forums. 
Like https://dnncommunity.org/forums/aft/982/afpg/2 mentioning "powered Web App resource (plan)"

**Comment by interfacemirror** [comment]
I have the same issue. Controls are missing in my installation folder. Tried downloading some of them from GitHub but Admin/Skins/ControlBar.ascx does not exist anywhere. Not sure what could be the resolution.

**Comment by patrickryan79** [comment]
This just happened to me with 9.10.0 on a clean install in my local testing environment.

Had to do this:

```
 UPDATE HostSettings
SET SettingValue = 'DesktopModules/admin/Dnn.PersonaBar/UserControls/PersonaBarContainer.ascx'
WHERE SettingName = 'ControlPanel';
```

I'm going to run through the process again to confirm. I may have done something wrong. Was in a hurry, as usual

**Comment by sleupold** [comment]
@patrickryan79 
did you check the previous setting value for the ControlPanel?
Did you use any custom template for site or install?

**Comment by patrickryan79** [comment]
@sleupold 

Yes, I think it was ... Admin/ControlBar.ascx or something like that.
No custom. all default

**Comment by sleupold** [comment]
thank you, I'll try to verify.

**Comment by valadas** [comment]
I am moving this issue to a discussion until it becomes actionable.

**Reply by trouble2** [reply]
Just installed a new v 10.3.3 on azure (P2V3) and had the same issue again.
With blank website / dutch.
I'd forgotten al about it until I found this issue I posted 5 years ago :-)


I just replaced '/Admin/ControlPanel/ControlBar.ascx' to 
'DesktopModules/admin/Dnn.PersonaBar/UserControls/PersonaBarContainer.ascx'
In the host settings table, and my install started working again....

**Comment by trouble2** [comment]
Just seen this in a new V10.3.3 installation again....
When I checked the Dotnetnuke.install.config.resources file in the install folder it still has an entry :
    <ControlPanel>Admin/ControlPanel/ControlBar.ascx</ControlPanel>

So I think this is the problem....
